main.js

```diff
@@ -49,36 +46,19 @@ function setupLogger(config) {
     var cfg_b = config.bunyan;
     assert.object(cfg_b, 'config.bunyan');
     assert.optionalString(cfg_b.level, 'config.bunyan.level');
-    assert.optionalObject(cfg_b.syslog, 'config.bunyan.syslog');
-
-    var level = LOG.level();
-
-    if (cfg_b.syslog && !LOG_LEVEL_OVERRIDE) {
-        assert.string(cfg_b.syslog.facility, 'config.bunyan.syslog.facility');
-        assert.string(cfg_b.syslog.type, 'config.bunyan.syslog.type');
-
-        var facility = bsyslog.facility[cfg_b.syslog.facility];
-        LOG = bunyan.createLogger({
-            name: NAME,
-            serializers: LOG_SERIALIZERS,
-            streams: [ {
-                level: level,
-                type: 'raw',
-                stream: bsyslog.createBunyanStream({
-                    name: NAME,
-                    facility: facility,
-                    host: cfg_b.syslog.host,
-                    port: cfg_b.syslog.port,
-                    type: cfg_b.syslog.type
-                })
-            } ]
-        });
-    }
 
     if (cfg_b.level && !LOG_LEVEL_OVERRIDE) {
         if (bunyan.resolveLevel(cfg_b.level))
             LOG.level(cfg_b.level);
     }
+
+    /*
+     * If the configured logging level is at or below DEBUG then enable
+     * source logging.
+     */
+    if (LOG.level() <= bunyan.DEBUG) {
+        LOG = LOG.child({src: true});
```

Comment:
FWIW, src:true is pretty heavy (CPU time). I'd generally be inclined to only turn it on when in *TRACE* level.

Reply:
Per chat conversation we're going to leave as-is for now and perhaps reevaluate if it becomes an issue.